### PR TITLE
Np 1968 delete draft doi handler

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

**Review by brinxmat — APPROVED** [review]

File: datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/DeleteDraftDoiAppEnv.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package no.unit.nva.datacite.handlers;
+
+import nva.commons.utils.Environment;
+import nva.commons.utils.JacocoGenerated;
+
+@JacocoGenerated
+public final class DeleteDraftDoiAppEnv {
```
Comment (axthosarouris) [inline datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/DeleteDraftDoiAppEnv.java:7]: All these methods  are used only in one handler? Why to have a separate class? each class we create is some extra time in the cold start

File: datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/DeleteDraftDoiAppEnv.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    public static final String DATACITE_CONFIG = "DATACITE_CONFIG";
```
Comment (axthosarouris) [inline datacite-delete-draft-doi-handler/src/main/java/no/unit/nva/datacite/handlers/DeleteDraftDoiAppEnv.java:9]: The two first env variables are not used.

File: datacite-delete-draft-doi-handler/src/test/java/no/unit/nva/datacite/handlers/DeleteDraftDoiHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import static nva.commons.utils.JsonUtils.objectMapper;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class DeleteDraftDoiHandlerTest {
```
Comment (axthosarouris) [inline datacite-delete-draft-doi-handler/src/test/java/no/unit/nva/datacite/handlers/DeleteDraftDoiHandlerTest.java:29]: what happens when deleting the draft doi fails?

**Review by axthosarouris — CHANGES_REQUESTED** [review]
Requested changes:
Remove env variables that are not in use.
Test that shows what happens when the remote server (datacite) returns a failure.

Suggested changes (but not required):
Delete the specialized environment class.

**Review by brinxmat — APPROVED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**